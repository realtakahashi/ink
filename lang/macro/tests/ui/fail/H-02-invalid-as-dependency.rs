use ink_lang as ink;

#[ink::contract(
    version = "0.1.0",
    compile_as_dependency = "yes",
)]
mod invalid_as_dependency {
    #[ink(storage)]
    struct InvalidAsDependency {}

    impl InvalidAsDependency {
        #[ink(constructor)]
        fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        fn message(&self) {}
    }
}

fn main() {}
