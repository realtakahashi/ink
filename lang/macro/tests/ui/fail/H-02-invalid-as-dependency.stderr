error: expected boolean literal
 --> $DIR/H-02-invalid-as-dependency.rs:5:29
  |
5 |     compile_as_dependency = "yes",
  |                             ^^^^^
